main.js

```diff
@@ -0,0 +1,115 @@
+
+    gameDirections.map((direction) => {
+      console.log(direction);
+    });
+  },
+
+  // Builds game board based on user input
+  startGame: function(numPegs, numDiscs) {
+    if (numPegs < 3 || numDiscs < 1) {
+      console.log('The number of pegs must be greater than or equal to 3 and the number of discs must be greater than 0! Please try again!');
+      return;
+    };
+
+    var pegBoard = Array(numPegs).fill().map(() => Array()); 
+
+    for (let discs = numDiscs; discs > 0; discs--) {
+      pegBoard[0].push(discs);
+    };
+
+    console.log(`- To move a disc, type into the console board.moveDisc(numberOne, numberTwo). numberOne is the peg you want to move from. numberTwo is disc you want to move the peg to. For example, board.moveDisc(1, 2) would move a disc from peg 1 to peg 2. Good luck!`);
+    this.printBoard(pegBoard);
+    board.startingPegBoard = pegBoard;
+    board.currentPegBoard = JSON.parse(JSON.stringify(pegBoard));
+  },
+
+  // Placeholders for game boards
+  startingPegBoard: [],
+  currentPegBoard: [],
+
+  // Moves discs between pegs
+  moveDisc: function (startPeg, endPeg) {
+    // Checks that startPeg and endPeg are between 1 - 3
+    if (startPeg < 1 || startPeg > this.numPegs || endPeg < 1 || endPeg > this.numPegs) {
+      console.log('Please enter a startPeg and/or endPeg value between 1 and 3.');
+      this.printBoard(this.currentPegBoard); 
+      return;
+    }
+
+    // Define starting and ending pegs
+    var currentPeg = startPeg - 1;
+    let endingPeg = endPeg - 1;
+    var startingPegValues = this.currentPegBoard[startPeg - 1];
+    var endingPegValues = this.currentPegBoard[endPeg - 1];
+    var startingPegLength = this.currentPegBoard[currentPeg].length;
+
+    // Get the value of the top discs on the starting and ending pegs
+    var discToMove = startingPegValues[startingPegValues.length - 1];
+    var endingTopDisc = endingPegValues[endingPegValues.length - 1];
+
+    // Sets an empty pegs top disc value to 0 if there are no discs
+    if (endingTopDisc === undefined) {
+      endingTopDisc = 0;
+    };
+
+    // Checks that there is a disc to move
+    if (startingPegLength <= 0) {
+      console.log('There is no disc to move on this peg. Please try again.');
+      this.printBoard(this.currentPegBoard);
+      return;
+    }
+
+    // Move the discs
+    if (discToMove < endingTopDisc || endingTopDisc === 0) {
+      this.currentPegBoard[endingPeg].push(discToMove);
+      this.currentPegBoard[currentPeg].splice(-1, 1);
+      this.checkWinner(this.currentPegBoard);
+    };
+
+    // Logs an error if user tries to put a larger disc on top of a smaller one
+    if (discToMove > endingTopDisc && endingTopDisc !== 0) {
+      console.log('You cannot move a larger disc on top of a smaller one, board is still:');
+      this.printBoard(this.currentPegBoard);
+    };
+  },
+
+  // Prints board to console
+  printBoard: function(pegBoard) {
+    pegBoard.map((peg, index) => {
```

Comment:
The callback function passed to `map` should return something because `map` is supposed to build a new array. You could replace `map` with `forEach` here and nothing would change.